lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -356,20 +360,22 @@ export class UserAgentApplication {
       scopes = this.filterScopes(scopes);
     }
 
-      this._loginInProgress = true;
       var idTokenObject;
-      idTokenObject= this.extratADALIdToken();
+      idTokenObject= this.extractADALIdToken();
       if (idTokenObject && !scopes) {
-          extraQueryParameters = Utils.constructUnifiedCacheExtraQueryParameter(idTokenObject, extraQueryParameters);
           this._logger.info("ADAL's idToken exists. Extracting login information from ADAL's idToken ");
+          var extraQueryParams = extraQueryParameters ? extraQueryParameters : "";
```

Comment:
can we directly pass extraQueryParameters and not this variable extraQueryParams. We can do the null check inside constructUnifiedCacheExtraQueryParameter